docroot/modules/humsci/hs_layouts/hs_layouts.install

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * @file
+ * Update functions for hs_layouts module.
+ */
+
+use Drupal\layout_builder\Section;
+
+/**
+ * Set main content id in content types default layout builder config.
+ */
+function hs_layouts_update_9401() {
+  $storage = \Drupal::entityTypeManager()->getStorage('entity_view_display');
+  $query = $storage->getQuery()
+    ->condition('targetEntityType', 'node')
+    ->condition('mode', 'default');
+  $items = $query->execute();
+  foreach ($items as $item) {
+    $view_display = $storage->load($item);
+    if ($view_display->isLayoutBuilderEnabled()) {
+      $sections = $view_display->getThirdPartySetting('layout_builder', 'sections');
+      if (count($sections)) {
+        $main_section = $sections[0];
+        if (count($sections) > 1) {
+          foreach ($sections as $section) {
+            if ($section->getLayoutSettings()['section_width'] !== 'hs-full-width') {
+              $main_section = $section;
+              break;
+            }
+          }
+        }
+        $main_section->setLayoutSettings(array_merge($main_section->getLayoutSettings(), ['main_content' => 'main-region']));
+        $view_display->save();
+      }
+    }
+  }
+}
+
+/**
+ * Set main content id in nodes with overridden layout builder config.
+ */
+function hs_layouts_update_9402() {
```

Comment:
This needs to be done in a batch process.

Reply:
Fixed.